src/main/ts/components/Editor.tsx

```diff
@@ -310,194 +230,310 @@ export class Editor extends React.Component<IAllProps> {
-    const editor = this.editor;
-    if (editor && editor.initialized) {
-      const newContent = editor.getContent();
-      if (this.props.value !== undefined && this.props.value !== newContent && this.props.rollback !== false) {
+  const handleEditorChange = React.useCallback((_evt: EditorEvent<unknown>) => {
+    const currentEditor = editorRef.current;
+    if (currentEditor && currentEditor.initialized) {
+      const newContent = currentEditor.getContent();
+      if (value !== undefined && value !== newContent && rollback !== false) {
         // start a timer and revert to the value if not applied in time
-        if (!this.rollbackTimer) {
-          this.rollbackTimer = window.setTimeout(
-            this.rollbackChange,
-            typeof this.props.rollback === 'number' ? this.props.rollback : 200
+        if (!rollbackTimerRef.current) {
+          rollbackTimerRef.current = window.setTimeout(
+            rollbackChange,
+            typeof rollback === 'number' ? rollback : 200
           );
         }
       }
 
-      if (newContent !== this.currentContent) {
-        this.currentContent = newContent;
-        if (isFunction(this.props.onEditorChange)) {
-          this.props.onEditorChange(newContent, editor);
+      if (newContent !== currentContentRef.current) {
+        currentContentRef.current = newContent;
+        if (isFunction(onEditorChange)) {
+          onEditorChange(newContent, currentEditor);
         }
       }
     }
-  };
+  }, [ value, onEditorChange, rollback, rollbackChange ]);
 
-  private handleEditorChangeSpecial = (evt: EditorEvent<KeyboardEvent>) => {
+  const handleEditorChangeSpecial = React.useCallback((evt: EditorEvent<KeyboardEvent>) => {
     if (evt.key === 'Backspace' || evt.key === 'Delete') {
-      this.handleEditorChange(evt);
+      handleEditorChange(evt);
     }
-  };
+  }, [ handleEditorChange ]);
 
-  private initialise = (attempts = 0) => {
-    const target = this.elementRef.current;
+  const initialise = React.useCallback((attempts = 0) => {
+    const target = elementRef.current;
     if (!target) {
       return; // Editor has been unmounted
     }
     if (!isInDoc(target)) {
-      // this is probably someone trying to help by rendering us offscreen
+      // is probably someone trying to help by rendering us offscreen
       // but we can't do that because the editor iframe must be in the document
       // in order to have state
       if (attempts === 0) {
         // we probably just need to wait for the current events to be processed
-        setTimeout(() => this.initialise(1), 1);
+        setTimeout(() => initialise(1), 1);
       } else if (attempts < 100) {
         // wait for ten seconds, polling every tenth of a second
-        setTimeout(() => this.initialise(attempts + 1), 100);
+        setTimeout(() => initialise(attempts + 1), 100);
       } else {
-        // give up, at this point it seems that more polling is unlikely to help
```

Comment:
I assume you did a find and replace for `this` and accidentally removed it from this comment too. The new comment doesn't make any sense so I'd suggest tweaking it or putting `this` back...